docs/index.adoc

```diff
@@ -117,6 +117,45 @@ class CheckSum implements Callable<Integer> {
 // The goal of the introduction is to show how easy picocli is.
 // (The rest of this document has plenty of details.)
 
+.Kotlin
+[[CheckSum-App-Kotlin]]
+[source,kotlin,role="secondary"]
+----
+package picocli.examples.kotlin
+
+import picocli.CommandLine
+import picocli.CommandLine.Command
+import picocli.CommandLine.Option
+import picocli.CommandLine.Parameters
+import java.io.File
+import java.math.BigInteger
+import java.nio.file.Files
+import java.security.MessageDigest
+import java.util.concurrent.Callable
+import kotlin.system.exitProcess
+
+@Command(name = "checksum", mixinStandardHelpOptions = true, version = ["checksum 4.5.1"],
+    description = ["Prints the checksum (MD5 by default) of a file to STDOUT."])
+class Checksum : Callable<Int> {
+
+    @Parameters(index = "0", description = ["The file whose checksum to calculate."])
+    var file = File("")
```

Comment:
Should the positional parameter have a default value of a non-existing file named `""`?
As an aside, generally I prefer default values to be specified in the annotation with `defaultValue = "xxx"` (in this example I used field initialization for the `algorithm` variable to simplify, but using the annotation has some advantages and is preferred).

I know it is not "Kotlin-esque", but shouldn't we do something like this instead:

```kotlin
@Parameters(index = "0" ...)
File file;
```

Reply:
```
@Parameters(index = "0" ...)
File file;
```

This doesn't compile: `Expecting member declaration`

I think the best/correct way to handle this is to mark the property with the `lateinit` modifier.

```
@Parameters(index = "0" ...)
lateinit var file : File
```

I changed the pull request accordingly.